### PR TITLE
CLI Enhancements

### PR DESCRIPTION
**Pull request by corylanou**
- Command history is now persistent to disk
- Added `help` command
- Fixed erorsr not printing out for invalid commands

### CODE REVIEW

**Comment by pauldix** [comment]
looks good. Is there a limit to how much history it saves?

File: cmd/influx/main.go [hunk 1]
```diff
@@ -83,56 +135,41 @@ func (c *cli) connect(cmd string) {
...
-		} else {
-			data, err = json.Marshal(i)
-		}
-		if err != nil {
-			fmt.Printf("ERR: %s\n", err)
-			return
-		}
-		fmt.Fprintln(os.Stdout, string(data))
+	var data []byte
+	if c.pretty {
```
Comment (otoolep) [inline cmd/influx/main.go]: Perhaps you don't need to do this -- you can just pass `pretty=true` on the URL.

**Comment by corylanou** [comment]
@pauldix 1000 by default

File: cmd/influx/main.go [hunk 2]
```diff
@@ -83,56 +135,41 @@ func (c *cli) connect(cmd string) {
...
+}
+
+func help() {
+	fmt.Println(`Usage:
+        connect <host:port>   connect to another node
+        pretty                toggle pretty print
+        use <db_name>         set current databases
+        exit                  quit the influx shell
+
+        show databases        show database names
```
Comment (otoolep) [inline cmd/influx/main.go]: There is nothing special about these, right? Just some useful help? I mean, there is no mention of `SELECT`.

**Comment by otoolep** [comment]
Some feedback, but +1.

File: cmd/influx/main.go (hunk 1 above)
Comment (corylanou) [inline cmd/influx/main.go]: @otoolep I'm not doing any HTTP in this CLI.  It's consuming our client library, which returns back a top level `influxdb.Results` object.  This is intentional so we dogfood our own client library.

**Comment by pauldix** [comment]
rad, :shipit:

File: cmd/influx/main.go (hunk 2 above)
Comment (corylanou) [inline cmd/influx/main.go]: @otoolep correct, nothing special.  I'm open to suggestions on what we think should be there.  I want to strike a balance between to much as I don't want to have this be in a state of flux if things change, but I do want it to give you some basic help.

At some point in the future, I'll leverage our system to actually do auto complete/suggestions.  but that is probably post 1.0.

**Pull request merged**